File: problems/506-relative-ranks/relative-ranks.swift


class Solution {
    func findRelativeRanks(_ score: [Int]) -> [String] {

    let sorted = score.sorted { $0 > $1 }
    var dict = [Int: String]()
    let ranking = ["Gold Medal","Silver Medal","Bronze Medal"]

    for i in (0 ..< score.count) {
        if (i < 3) {
            dict[sorted[i]] = ranking[i];
        } else {
            dict[sorted[i]] = String(i + 1);
        }
    }

    return score.compactMap { dict[$0] }
    // return score.map(el => map[el])
    }
}

File: problems/506-relative-ranks/relative-ranks.js
/**
 * @param {number[]} score
 * @return {string[]}
 */
const findRelativeRanks = (score) => {
    let result = [];
    const sorted = [...score].sort((a, b) => b - a);
    for (let i = 0; i < score.length; i++) {
        if (score[i] === sorted[0]) {
            result = [...result, "Gold Medal"];
        } else if (score[i] === sorted[1]) {
            result = [...result, "Silver Medal"];
        } else if (score[i] === sorted[2]) {
            result = [...result, "Bronze Medal"];
        } else {
            const rank = sorted.indexOf(score[i]) + 1;
            result = [...result, String(rank)];
        }
    }
    return result;
};